projects/kit/components/input-number/input-number.component.ts

```diff
@@ -117,7 +117,8 @@ export class TuiInputNumberComponent
             (this.decimal !== 'never' &&
             this.nativeValue.includes(this.numberFormat.decimalSeparator)
                 ? this.precision + 1
-                : 0)
+                : 0) +
+            this.numberFormat.thousandSeparator.repeat(5).length
         );
```

Comment:
can we split into different variables to make it easier to read?)

```ts
 get calculatedMaxLength(): number {
   const decimalPart = this.decimal !== 'never' && this.nativeValue.includes(this.numberFormat.decimalSeparator);
   const precision = decimalPart ? this.precision + 1 : 0;
   const takeThousand = this.numberFormat.thousandSeparator.repeat(5).length;

   return DEFAULT_MAX_LENGTH + precision + takeThousand;
}
```

I'm not sure about variable names, but what do you think?